src/node/transaction.cpp

```diff
@@ -36,10 +36,18 @@ TransactionError BroadcastTransaction(NodeContext& node, const CTransactionRef t
         if (!existingCoin.IsSpent()) return TransactionError::ALREADY_IN_CHAIN;
     }
     if (!node.mempool->exists(hashTx)) {
-        // Transaction is not already in the mempool. Submit it.
+        // Transaction is not already in the mempool.
         TxValidationState state;
-        if (!AcceptToMemoryPool(*node.mempool, state, std::move(tx),
-                nullptr /* plTxnReplaced */, false /* bypass_limits */, max_tx_fee)) {
+        CAmount fee{0};
+        if (AcceptToMemoryPool(*node.mempool, state, tx,
```

Comment:
I think this approach is okay for now but it would be better to refactor further in the future. For now the mempool is guaranteed not to change due to tacking `cs_main` lock above but if we remove this lock requirement from ATMP in the future you might have different views of the mempool in the future, thus being rejected at first try and not being evaluated against `max_tx_fee` but succeeding at second throw and effectively bypassing the protection.

So technically the comment "Transaction fee is acceptable" is misleading as fee check might have been skipped.

You could add a new function `GetTransactionFee` composed of both a mempool lock tacking, a outpoints fetching logic (`src/validation.cpp` L649) and `Consensus::CheckTxInputs`.